client.go

```diff
@@ -197,6 +202,63 @@ func (e *FailedResponse) Error() string {
 	return fmt.Sprintf("%v: %s", e.StatusCode, string(e.Response))
 }
 
+// Patch updates a resource for the target URL
+func (c *Client) Patch(url string, contentType ContentType, body interface{}, dest interface{}) error {
+	var err error
+	jsonBody, err := json.Marshal(body)
+
+	if err != nil {
+		return fmt.Errorf("could not patch, body could not be marshalled to json: %w", err)
+	}
+
+	req, err := http.NewRequest("PATCH", url, bytes.NewBuffer(jsonBody))
+
+	if err != nil {
+		return err
+	}
+
+	if c.FiatUser != "" {
+		req.Header.Set(SpinFiatUserHeader, c.FiatUser)
+		req.Header.Set(SpinFiatAccountHeader, c.FiatUser)
+	}
+
+	req.Header.Set("Content-Type", string(contentType))
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode >= 200 && resp.StatusCode < 400 {
+		if err != nil {
+			return err
+		}
+
+		// Only unmarshalling if we actually have a response body (as opposed
+		// to, for example, a simple "201 Created" response)
+		if len(b) > 0 {
+			err := json.Unmarshal(b, &dest)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+	} else if resp.StatusCode >= 400 && resp.StatusCode < 600 {
```

Comment:
if len(b) == 0 -> return ErrUnsupportedStatusCode. Probably not what we want.

Reply:
I don't like `ErrUnsupportedStatusCode` generally, I feel like if we're gonna refactor we should Do It Right™